src/signing_key.rs

```diff
@@ -106,6 +110,55 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> From<PrivateKeyInfo<'a>> for SigningKey {
+    fn from(pki: PrivateKeyInfo) -> SigningKey {
+        pki.private_key.try_into().expect("Ed25519 private key wasn't 32 bytes")
+    }
+}
+
+impl ToPrivateKey for SigningKey {
+    fn to_pkcs8_der(&self) -> PrivateKeyDocument {
+        let alg_info = AlgorithmIdentifier {
+            oid: ObjectIdentifier::new("1.3.101.112"), // RFC 8410
```

Comment:
If you add algorithm OID validation as noted above, you might want to hoist this OID into a constant (i.e. defined at the top of the module):

```rust
const OID = ObjectIdentifier::new("1.3.101.112");
```